vuu-ui/packages/vuu-shell/src/theme-provider/ThemeProvider.tsx

```diff
@@ -48,27 +62,30 @@ const createThemedChildren = (
 };
 
 interface ThemeProviderProps {
-  children: ReactElement;
+  children?: ReactElement;
   density?: Density;
-  theme?: string;
-  applyClassesToChild?: true;
+  // theme?: Theme;
+  themeMode?: ThemeMode;
+  applyClassesTo?: TargetElement;
 }
 
 export const ThemeProvider = ({
   children,
-  density: densityProp,
-  theme: themeProp,
+  // theme: themeProp,
+  themeMode: themeModeProp,
+  density: densityProp
 }: ThemeProviderProps) => {
-  const { theme: inheritedTheme, density: inheritedDensity } =
-    useContext(ThemeContext);
+  const [themeMode, setThemeMode] = useState<ThemeMode>("light");
+  const [density, setDensity] = useState<Density>("high");
 
-  const density = densityProp ?? inheritedDensity ?? DEFAULT_DENSITY;
-  const theme = themeProp ?? inheritedTheme ?? DEFAULT_THEME;
-
-  const themedChildren = createThemedChildren(children, theme, density);
+  const currentDensity = densityProp ?? density ?? DEFAULT_DENSITY;
+  const currentThemeMode = themeModeProp ?? themeMode ?? DEFAULT_THEME_MODE;
```

Comment:
if this was the correct behaviour, then a cleaner way to implement this would simple be like this
```
export const ThemeProvider({
   themeMode = DEFAULT_THEME_MODE
})
```
then just use `themeMode` in the call below. However, this isn't correct, so you can't do it that way. We're missing the (possible) inherited value, so it has to be done like the original code